Fixing return error on run command

rajasec opened the pull request:
When we run sudo ctr run command
Usage is saying about IMAGE ID
USAGE:
   ctr run [command options] IMAGE ID [COMMAND] [ARG...]
When we run 
sudo ctr run
Error is returning as 
ctr: container id must be provided
As ctr takes image ID as argument, changed the error type to image id instead of container id

Signed-off-by: rajasec <rajasec79@gmail.com>

estesp commented:
This change is correct, but the code is still confusing to the casual reader IMO. Everywhere else in the code the "IMAGE" is called a 'ref' or 'imageRef'. I think it would be good to change `id` in the code to `imageRef` so it isn't as confusing.  The "ID" is read in `newContainer` and there both `ref` and `id` are used to differentiate image (id) reference and container id

rajasec commented:
@estesp 
Let me know if I need to change IMAGE ID -> IMAGE REF, so that I re-submit the changes

estesp commented:
actually..that's a separate confusing thing.. "IMAGE" and "ID" are 2 totally different things.. `IMAGE` == image id/image ref (.e.g. `docker.io/library/alpine:latest`) and `ID` == the ID name of the container being run. So those are two different parameters on the command line.

The help string would get longer to be clearer but it could be changed from `IMAGE ID` -> `IMAGE_REF CONTAINER_ID` to clear up that they are two unique parameters to `ctr run`

rajasec commented:
@estesp 
I have made the required changes, please have an look. I have added separate error conditions for image ref and container ID and also updated the usage too.

estesp commented:
This looks like reasonable behavior to me:
```
((bd2f684...)) [1]> bin/ctr run 
ctr: image ref must be provided
((bd2f684...)) [1]> bin/ctr run fred 
ctr: container id must be provided
```

Anyone else have an opinion? @stevvooe @mlaventure

stevvooe commented on cmd/ctr/run.go:
```diff
@@ -94,10 +93,14 @@ var runCommand = cli.Command{
 		)
 		defer cancel()
 
+		if imageRef == "" {
```
Where is this used?

estesp replied:
it isn't here; `newContainer()` in *cmd/ctr/run_unix.go* uses the ref without any checking for empty string.

stevvooe commented:
Reject

I would prefer `IMAGE CONTAINER`. There is no need here for compound naming.

estesp commented:
That's fine @stevvooe ; @rajasec could you update the help text with Stephen's suggestion? Thanks and sorry for spinning a bit on this!

rajasec commented:
@estesp @stevvooe 
Updated the usage.

dmcgowan commented on cmd/ctr/run.go:
```diff
@@ -2,15 +2,14 @@ package main
 	specs "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/pkg/errors"
 	"github.com/urfave/cli"
+	"syscall"
```
The empty line that was there previously made goimports do the right thing and separate std library imports from non-standard imports.

stevvooe commented:
LGTM after an import fix.

estesp commented:
@rajasec can you move that import back to it's original location? Sounds like after that we can merge

rajasec commented:
@estesp 
Reverted back the imports to original.
